### PR TITLE
Gang Fighting Styles

### PR DESCRIPTION
**Pull request by ghost**
This PR introduces gang fighting styles. Certain gangs will be able to purchase different equipment. For instance, the Sleeping Carp gang focuses around ancient and modern martial arts rather than using ballistic weapons. The system is modular, and one can introduce their own fighting styles (for instance a Cuban fighting style that allows you to purchase custom bombs).

### CODE REVIEW

**Comment by pudl** [comment]
I think this is adding more not really useful nuances to gang. I don't like it. :-1:

**Comment by Ikarrus** [comment]
I don't have a problem with adding martial arts to gang, but perhaps there's a better way to get it than the very small RNG chance of getting the Sleeping Carp gang?

And is it really that OP that it needs 50 influence?

**Comment by xxalpha** [comment]
Maybe instead of one gang getting the martial arts, you could have all gangs in the round be of the same style and they all have martial arts. A chance for gang mode to be chinese gang mode.

**Comment by Ikarrus** [comment]
One idea I had when someone in coderbus pitched the idea was to tie it into the gang outfits wearing them suddenly made you jacky chan or whatever. 

Then again I don't know how OP or UP the martial arts is without playing but from what I understand it's just hitting people with attacks in a certain order right?

Can't be that bad.

**Comment by Anonus** [comment]
Can't be that bad. You can gib people with plasmafist if you chain that shit right.

**Comment by Thunder12345** [comment]
Should be alright as long as there's no plasma fist. Plasma fist is a five hit gib.

**Comment by Ikarrus** [comment]
It just makes me think of this 

https://www.youtube.com/watch?v=PDQPIJSO0XA

**Comment by Anonus** [comment]
It'd probably be fine if it was a version that just instantly killed you instead, and maybe renamed to something. Like, 'secret carp technique' or something.

**Comment by Xhuis** [comment]
I'm currently working on a martial art style that is in effect by wearing a special suit. I am trying to get the big carp onesie that CosmicScientist made so I can use that.

**Comment by PKPenguin321** [comment]
lmao, plasma fist scrolls? you know you can throw people down entire hallways, stun people in a radius, and gib people with just your fists with that, right?

**Comment by Iamgoofball** [comment]
@Ikarrus I fucking wish it was like that

File: code/datums/martial.dm [hunk 1]
```diff
@@ -319,6 +319,90 @@
...
+		H << "<span class='notice'>You read through the scroll and realize you know the secrets of the Plasma Fist.</span>"
+		if(istype(src, /obj/item/weapon/plasma_fist_scroll/oneuse))
+			H.drop_item()
+			visible_message("<span class='warning'>[src] lights up in fire and quickly burns to ash.</span>")
+			new /obj/effect/decal/cleanable/ash(get_turf(src))
+			qdel(src)
 		used = 1
 		desc += "It looks like it's magic was used up."
+
+/obj/item/weapon/plasma_fist_scroll/oneuse
```
Comment (RemieRichards) [inline code/datums/martial.dm]: either you've implemented this completely crappily, or this type name is incredibly wrong.

**Comment by Jalleo** [comment]
@ikarrus only if wearing a 1920's suit gives you double the ammunition of any gun you use fpr those shooting executions in maint

File: code/datums/martial.dm (hunk 1 above)
Comment (Anonus) [inline code/datums/martial.dm]: It is entirely correct, as plasmafist scrolls are already one use.

Comment (RemieRichards) [inline code/datums/martial.dm]: You've just proved me correct, not incorrect.
the scrolls are already one use, so to define a subtype that claims to be "more one use than one use"... is nonsensical and WRONG.

Comment (Xhuis) [inline code/datums/martial.dm]: This is removed in the next commit

**Comment by Xhuis** [comment]
New martial art style added

**Comment by Xhuis** [comment]
To be honest this is basically ready for merging unless any maintainers have any qualms
@Ikarrus

**Comment by Ikarrus** [comment]
I still don't like the RNG factor of getting martial arts as a gang, as well as this complicating an already complicated game mode, but I guess I have no real objections to this. 

I'll have to see this in action in game to really decide.

**Comment by PKPenguin321** [comment]
yeah the RNG could be annoying, if anything let the boss choose at roundstart or something

**Comment by Ikarrus** [comment]
Hey on the bright side keeping it rare will give us time to work it out if it ends up being terrible.

**Comment by Xhuis** [comment]
I wanted to keep it rare until we know it's balanced enough to add more gangs for.

**Comment by Ikarrus** [comment]
Sounds gud

**Comment by Razharas** [comment]
Gang is Ikarrus baby, ill let him have a final word of yay or nay on this

**Comment by Razharas** [comment]
@Ikarrus

**Comment by Razharas** [comment]
Ikarrus said its ok

**Comment by Xhuis** [comment]
Nice, let me know how it works out on the servers.

**Pull request merged**